developer/src/kmc-kmn/src/ng-compiler/lexer.ts

```diff
@@ -0,0 +1,212 @@
+
+
+    // we cannot handle line continuation if emitAll is true
+    // (i.e. emitAll:true => handleContinuation:false)
+    if (emitAll) {
+      handleContinuation = false;
+    }
+
+    // loop over all ScanRecognizers looking for a match at the offset into the buffer
+    while (!(iterResult = patternIterator.next()).done && !tokenMatch) {
+      const recognizer: ScanRecognizer    = iterResult.value;
+      recognizer.regExp.lastIndex         = this.offset;
+      const match: RegExpExecArray | null = recognizer.regExp.exec(this.buffer);
+
+      if (match) {
+        this.line = this.line.concat(match[0].toString());
+        let line: string = null;
+        if (handleContinuation) {
+          // if handleContinuation is true, no CONTINUATIONs will be emitted,
+          // nor will NEWLINEs that follow CONTINUATIONs be emitted
+          if (recognizer.tokenType === TokenType.CONTINUATION) {
+            this.seenContinuation = true;
+          } else if (recognizer.tokenType === TokenType.NEWLINE) {
+            if (!this.seenContinuation) {
+              if (emitAll || recognizer.emit) {
+               this.tokenList.push(new Token(recognizer.tokenType, match[0], this.lineNum, this.charNum, this.line, this.filename));
+              }
+              this.line = '';
+            }
+            this.seenContinuation = false;
+          } else { // other tokens
+            if (emitAll || recognizer.emit) {
+              this.tokenList.push(new Token(recognizer.tokenType, match[0], this.lineNum, this.charNum, null, this.filename));
+            }
+          }
+        } else { // not handling continuation
+          if (recognizer.tokenType === TokenType.NEWLINE) {
+            line      = this.line;
+            this.line = '';
+          }
+          if (emitAll || recognizer.emit) {
+            this.tokenList.push(new Token(recognizer.tokenType, match[0], this.lineNum, this.charNum, line, this.filename));
+          }
+        }
+        tokenMatch  = true;
+        this.offset = recognizer.regExp.lastIndex; // advance the buffer offset past the matched token
+        if (recognizer.tokenType === TokenType.NEWLINE) {
+          this.lineNum += 1;
+          this.charNum  = 1;
+        } else {
+          this.charNum += match[0].length;
+        }
+      }
+    }
+
+    // discard matching empty brackets
+    // Empty brackets are simply discarded by the current compiler, and allowing them through to the syntax analyser
+    // hugely complicates the grammar. When the Lexer is extended for error handling, it may be that a few more token
+    // combinations can be identified and warnings/errors given. Possibilities include mismatched braces, square
+    // brackets etc. I included this one at this stage because of an instance in the keyboard repository
+    // https://github.com/keymanapp/keyboards/blob/92240aaf75261aa771a019d0eee59a4e58806644/release/sil/sil_senegal_bqj_azerty/source/sil_senegal_bqj_azerty.kmn#L746
```

Comment:
I consider that to be a syntax error in the existing keyboard and I would be okay with the new compiler flagging it as such, even if it is missed by the old compiler. If that simplifies things here, all the better.